MoreLinq.Test/PadLeftTest.cs

```diff
@@ -0,0 +1,132 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using NUnit.Framework;
+    using System;
+    using System.Linq;
+    using System.Collections.Generic;
+    
+
+    [TestFixture]
+    public class PadLeftTest
+    {
+        // PadLeft(source, width)
+
+        [Test]
+        public void PadLeftNegativeWidth()
+        {
+            Assert.ThrowsArgumentException("width",() => new int[0].PadLeft(-1));
+        }
+
+        [Test]
+        public void PadLeftIsLazy()
+        {
+            new BreakingSequence<int>().PadLeft(0);
+        }
+
+        [Test]
+        public void PadLeftWideSourceSequence()
```

Comment:
Turn `PadLeftWideSourceSequence`, `PadLeftEqualSourceSequence` and `PadLeftNarrowSourceSequence` into test cases? Same for when testing the other overloads.